### ISSUE TITLE
Support multi indexing

### ISSUE THREAD

**Issue by jdoliner** [rethinkdb/rethinkdb#933]
@linuxdynasty requested on IRC being able to index in such a way that each document can have multiple secondary keys in the same secondary index. This would allow you to query data that looks like the following:

```
{fruits : ['apple', 'orange', 'tomato']}
{fruits : ['papaya', 'cherry', 'sandwich']}
```

With something like:

``` Python
r.table("fruit_stands").get_all('apple', index="fruits_sold")
```

---

This is actually very easy to do with the way secondary indexes are implemented. I think the biggest question is actually how we'd represent this in the query language. Off the top of my head I'd like to have a syntax like so:

```
r.table("foo").index_create("fruits_sold", lambda x: r.multi_index(x["fruits"]))
```

Basically `r.multi_index` is a specially term which takes an array and just indicates to the secondary index code that it should index by each element of the array individually rather than indexing by the whole array.
